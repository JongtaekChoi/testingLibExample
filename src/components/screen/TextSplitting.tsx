import React, { ReactElement, useState } from 'react';

import { DefaultNavigationProps } from '../../types';
import styled from 'styled-components/native';
import { Alert, TouchableOpacity } from 'react-native';
import Button from '../shared/Button';

const Container = styled.View`
  flex: 1;
  background-color: transparent;
  align-items: center;
  padding: 15px;
`;

const StyledTextInput = styled.TextInput`
  font-size: 16;
  color: blue;
  height: 200px;
  border-width: 0.5px;
  width: 100%;
`;

const StyeldText = styled.Text`
  font-size: 16;
  color: blue;
`;

interface Props {
  navigation: DefaultNavigationProps<'TextSplitting'>;
}

function Page(props: Props): ReactElement {
  const [ textValue, setTextValue ] = useState<string>(''); 
  const [ sentenseList, setSentenseList ] = useState<string[]>([''])
  return (
    <Container>
      <StyledTextInput
        testID="textInput"
        onChangeText={(text) => {
          setTextValue(text);
          setSentenseList(text.split(/[.!?]/));
        }}
        value={textValue}
        multiline
        />
      { 
        sentenseList.map((sentense: string): ReactElement => 
          (<StyeldText>{sentense.trim()}</StyeldText>)) 
      }
      <Button 
        testID="removeBtn"
        onClick={() => {
          Alert.alert('Remove All','',[
            {
              onPress: () => {
                setTextValue('')
                setSentenseList([])
              }
            },
          ])
        }}
        text="Remove All"
      />
    </Container>
  );
}

export default Page;
