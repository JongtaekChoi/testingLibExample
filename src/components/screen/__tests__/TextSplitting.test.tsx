import 'react-native';

import React, { ReactElement } from 'react';
import { RenderResult, act, fireEvent, render } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import TextSplitting from '../TextSplitting';
import renderer from 'react-test-renderer';

let props: any;
let component: ReactElement;
let testingLib: RenderResult;

jest.mock('Alert', () => {
  return {
    alert: (title, message, buttons) => {
      buttons[0].onPress();
    }
  }
});

describe('[TextSplitting] screen', () => {
  beforeEach(() => {
    props = createTestProps({});
    component = createTestElement(<TextSplitting {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('interactions', () => {
    const input = testingLib.getByTestId('textInput');
    act(() => {
      fireEvent.changeText(input, 'I have a heart, too, HwaLan. \
      If you trample my heart like this, then I\'ll be a gangster! \
      Do I kidnap you like a gangster?');
    });
    const btn = testingLib.queryByTestId('removeBtn');
    act(() => {
      fireEvent.press(btn);
    });
  });
});
