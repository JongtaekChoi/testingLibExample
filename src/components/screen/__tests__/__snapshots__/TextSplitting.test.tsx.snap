// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TextSplitting] screen renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderWidth": 0.5,
          "color": "blue",
          "fontSize": 16,
          "height": 200,
          "width": "100%",
        },
      ]
    }
    testID="textInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 16,
        },
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="removeBtn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#069ccd",
            "borderColor": "#069ccd",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 52,
            "justifyContent": "center",
            "width": 320,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 14,
            },
          ]
        }
      >
        Remove All
      </Text>
    </View>
  </View>
</View>
`;
